src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -275,6 +276,21 @@
 )
 
 
+def _ensure_ascii(json_string: str) -> str:
+    """Ensure a json string only contains ascii characters.
+
+    We need to use this when sending data to fastapi, otherwise it fails to
+    parse the request body.
+
+    Args:
+        json_string: A json string
+
+    Returns:
+        The input string with only ascii characters.
+    """
+    return json.dumps(json.loads(json_string), ensure_ascii=True)
```

Comment:
![CleanShot 2024-08-08 at 13 50 50@2x](https://github.com/user-attachments/assets/5c967a5a-6c1c-4155-aa0d-9d647a1baccd)

Is this what we want? Should we warn people when they're doing this?

Reply:
If you load this into a pydantic model, it parses it back into café:

![Screenshot 2024-08-08 at 13 54 25](https://github.com/user-attachments/assets/f95506e9-d418-4bc7-9293-05a2b9fceaa7)

Reply:
I also tried with pydantic1, where `model.json()` did the ascii encoding automatically and we never had issues there right?

Reply:
Sorry I should have been clearer. I wanted to say that this seems like it'd be unexpected for these values to be transformed this way. Wondering if we want an approach that replaces the letters maybe with some `?` character or something instead?

```python
ascii_text = text.encode('ascii', errors='replace').decode('ascii')
print(ascii_text)  # Output: caf?
```

There's also a bunch of packages that handle this, obv, like `unidecode`. Probably overkill here, but it'd at least turn `café` into `cafe`.

Reply:
This is only for requests we send to our own endpoints, in which the request will get converted to a pydantic model again, which means `café` will stay `café`.